src/puppetlabs/puppetdb/jdbc.clj

```diff
@@ -36,10 +36,10 @@
   string via (apply str v)."
   [& commands]
   (sql/db-do-commands *db* true
-                      (for [c commands]
-                        (if (vector? c)
-                          (apply str c)
-                          c))))
+                      (if (not-any? vector? commands)
```

Comment:
Originally was just half-thinking we might avoid the work if the input is fine, but even if that were plausible, it doesn't really support with my "let's always make it a vector" intent very well.